src/lib/array.jl

```diff
@@ -80,26 +80,20 @@ _droplike(dy::Union{LinearAlgebra.Adjoint, LinearAlgebra.Transpose}, dxv::Abstra
 
 for f in [push!, pop!, pushfirst!, popfirst!]
   @eval @adjoint! $f(xs, x...) =
-    push!(xs, x...), _ -> error("Mutating arrays is not supported")
+    $f(xs, x...), _ -> error("Mutating arrays is not supported")
 end
 
-# This is kind of bad, but at least we don't materialize the whole
-# array. Prefer to use `Buffer`
-# function _pullback(cx::Context, ::typeof(push!), xs::AbstractVector{<:AbstractArray}, x::AbstractArray{T}...) where T
-@adjoint! function push!(xs::AbstractVector{<:AbstractArray}, x::AbstractArray{T}...) where T
-  sz_xs = size.(xs)
-  sz_x = size.(x)
-  push!(xs, x...), Δ -> begin
-    (Δ, map(x -> Ones{T}(x...), sz_x)...)
+# Exceptions:
+@adjoint! function push!(dst::AbstractVector, xs::AbstractArray...)
```

Comment:
```suggestion
@adjoint! function push!(dst::AbstractVector{<: AbstractArray}, xs::AbstractArray...)
```

Reply:
Also retain the comments.

Reply:
Tests will fail if you do this. If you think it too wide, then can you invent a test case which it gets wrong?

Reply:
It'll be a memory hog otherwise and be slow.